src/compiler/checker.ts

```diff
@@ -6568,6 +6616,394 @@ module ts {
             }
         }
 
+        function checkJsxElement(node: JsxElement) {
+            if ((compilerOptions.jsx || JsxEmit.None) === JsxEmit.None) {

```

Comment:
This condition looks wrong. Did you mean `if (compilerOptions.jsx === JsxEmit.None)`